### ISSUE TITLE
Evaluate AWSGraphQLSubscriptionOperationCancelTests for cause of sporadic failures in dry runs

### ISSUE THREAD

**Issue by brennanMKE** [aws-amplify/amplify-ios#1305]
**Describe the bug**

These tests have failed multiple times recently possibly due to a race condition. Temporarily the timeout could be increased but if it is a race condition it is possible even more time would still result in tests failing. These tests will need to be evaluated to reproduce these sporadic failures locally to observe the stack frames when the tests fail and determine what can be done to fix the problem.

**To Reproduce**
1. Run these tests repeatedly (use performance loop features of XCTest)
2. Run until tests fail like in dry runs
3. Analyze the stack frames to see where threads are blocked
4. Determine how to eliminate the race condition

**Expected behavior**

These tests use expectations which are fulfilled so they should not fail based on timing. They should run quickly and fulfill the expections before the timeout expires.

**Screenshots**

**Environment(please complete the following information):**
 - Amplify Framework Version: [e.g. 1.5.3]
 - Amplify Categories:
   - [ ] Analytics
   - [ ] API
   - [ ] Auth
   - [x] DataStore
   - [ ] Predictions
   - [ ] Storage
 - Dependency Manager: [e.g. SwiftPM, Cocoapods]
 - Swift Version: [e.g. 5.0]
 - CLI Version: [e.g. 4.41.2, Run `amplify version`]
 - Xcode Version: [e.g. 12.4 (12D4e), Run `xcodebuild -version`]
 - Include any relevant log output under `~/.amplify/logs/amplify-cli-<issue-date>.log`

**Is this a regression (i.e. was this working before a version upgrade?)
*If applicable provide the versions used before and after the upgrade.*

**Device Information (please complete the following information):**
 - Device: [e.g. iPhone6, Simulator]
 - iOS Version: [e.g. iOS 11]
 - Specific to simulators:

**Additional context**

**Comment by brennanMKE** [comment]
* [Failing dry run](https://app.circleci.com/pipelines/github/aws-amplify/amplify-ios/3416/workflows/40dc390d-726a-49be-9b2b-aa5f57613640/jobs/29406)

`testCancelSendsCompletion`

    > Asynchronous wait failed: Exceeded timeout of 0.3 seconds, with unfulfilled expectations: "Received value for disconnected".

**Comment by ameter** [comment]
I haven't been able to reproduce this failure locally or on CircleCI.  I added a performance measure block to the test and ran it several thousand times locally.  I also pushed a branch with the modified test and it ran successfully 1,000 times on CircleCI.  https://github.com/aws-amplify/amplify-ios/blob/chore/testAWSGraphQLSubscriptionOperationCancelTests/AmplifyPlugins/API/AWSAPICategoryPluginTests/Operation/AWSGraphQLSubscriptionOperationCancelTests.swift

Looking into this a little more, I noted that a subset of CircleCI's macOS fleet was undergoing emergency maintenance at the time of the failed test.  It's possible that this contributed to the failure:
https://status.circleci.com/incidents/p6km16s326m5

This emergency maintenance occurred multiple times from 28-30 June.
https://status.circleci.com/history

Closing this issue for now.  We should re-open if these failures happen again.
